File: echoServers/echo.go
package main

import (
	"bufio"
	"fmt"
	"net"
)

func main() {
	server, err := net.Listen("tcp", ":8080")
	if err != nil {
		panic(err)
	}

	fmt.Println("Listening on", server.Addr())

	for {
		go func(conn net.Conn, err error) {
			for {
				if err == nil {
					in := bufio.NewReader(conn)
					message, err := in.ReadString('\n')
					if err == nil {
						_, err = fmt.Fprint(conn, message)
						if err != nil {
							conn.Close()
							return
						}
					} else {
						conn.Close()
					}
				} else {
					conn.Close()
					return
				}
			}
		}(server.Accept())
	}
}

File: echoServers/echo.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.PrintWriter;
import java.net.ServerSocket;
import java.net.Socket;

public class TestServer {
    public static void main(String[] args) throws IOException {
        new TestServer().start(new ServerSocket(8080));
    }

    @SuppressWarnings("InfiniteLoopStatement")
    private void start(ServerSocket sock) throws IOException {
        while (true) {
            Socket conn = sock.accept();
            new Thread(() -> {
                try {
                    String message = new BufferedReader(new InputStreamReader(conn.getInputStream())).readLine();
                    new PrintWriter(conn.getOutputStream(), true).println(message);
		    conn.close();
                } catch (IOException e) {
                    System.err.println(e.toString());
                }
            }).start();
        }
    }
}
